File: Application/Sources/Helpers/Categories/SRGChannel+PlaySRG.m
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

#import "SRGChannel+PlaySRG.h"

#import "ApplicationConfiguration.h"

@import libextobjc;

@implementation SRGChannel (PlaySRG)

- (UIImage *)play_largeLogoImage
{
    if (self.transmission == SRGTransmissionRadio) {
        return RadioChannelLargeLogoImage([ApplicationConfiguration.sharedApplicationConfiguration radioChannelForUid:self.uid]);
    }
    else {
        return TVChannelLargeLogoImage([ApplicationConfiguration.sharedApplicationConfiguration tvChannelForUid:self.uid]);
    }
}

@end

File: Application/Sources/Helpers/Extensions/SRGChannel+PlaySRG.swift
//
//  Copyright (c) SRG SSR. All rights reserved.
//
//  License information is available from the LICENSE file.
//

import SRGDataProviderModel

extension SRGChannel {
    var play_logoImage: UIImage? {
        if transmission == .radio {
            let radioChannel = ApplicationConfiguration.shared.radioChannel(forUid: uid)
            if ApplicationConfiguration.shared.isAudioContentHomepagePreferred {
                return RadioChannelSquareLogoImage(radioChannel)
            } else {
                return RadioChannelLogoImage(radioChannel)
            }
        } else {
            return TVChannelLogoImage(ApplicationConfiguration.shared.tvChannel(forUid: uid))
        }
    }

    var play_largeLogoImage: UIImage? {
        if transmission == .radio {
            let radioChannel = ApplicationConfiguration.shared.radioChannel(forUid: uid)
            if ApplicationConfiguration.shared.isAudioContentHomepagePreferred {
                return RadioChannelSquareLargeLogoImage(radioChannel)
            } else {
                return RadioChannelLargeLogoImage(radioChannel)
            }
        } else {
            return TVChannelLargeLogoImage(ApplicationConfiguration.shared.tvChannel(forUid: uid))
        }
    }
}
